### DISCUSSION TITLE
Promise Ring, Random Topics, Jokes and Related Stuff. Any Thoughts? (Spoilers)

### DISCUSSION THREAD

**Discussion by Justformas** [Monika-After-Story/MonikaModDev#9124] [general]
@Nikv2005 

I wanted to continue something I started talking about in issue #9070 and also in another discussion.

Since then, curiosity got the best of me on this. I had already unlocked the Piano and even reached 600 affection, but still no Promise Ring topic. I don't have random chatter set to repeat and usually I have the frequency set to normal or often. But personally, I'm not the type that often likes to "brute force" more random topics by sitting around doing nothing. Only sometimes when there's nothing else I want to do. Otherwise I just like them to sort of come naturally, which isn't that often since I'm usually doing something else. I suspect there's still quite a few random topics that I haven't gotten yet.

So I decided what the heck, I'll just try giving her the Promise Ring, and yes it works even if you haven't seen the topic. But I wasn't going to keep it that way; I decided I was going to restore a backup. But before I restored it, and because I am how I am, shortly after giving her the Promise Ring, I decided to hit Monika with "I'm breaking up with you", "Do you want to meet my girfriend?" and yes, even "You're a murderer". I can seem so evil can't I? Except, I wasn't actually planning on saying all of those things on the same day as giving her the Promising Ring, but then I thought what the heck, I'm restoring a backup anyway. There were no special responses or extra affection loss for any of those things in this circumstance, but I didn't really expect there to be. It would be highly situational and most people probably wouldn't even experience them. But now my curiosity is satisfied on this, so I probably won't say any of those things at all on the same day that I actually want to proceed with giving her the Promise Ring. That's even better.

So, any thoughts on the whole random topic thing? I know some topics go straight to "Unseen" for you to choose once the conditions are met, but those are things you initiate like questions to Monika, rather than random topics which are the other way around (at first). However, should some things like the Promise Ring topic be something you can choose to initiate her talking about at a certain point? Or should it be prioritized higher for a random topic (if that's possible)? There doesn't seem to be any other way to find out about it in game, so it seems like a pretty important/crucial topic.

**Comment by Nikv2005** [comment]
Promise ring topic is random so if you don't see it , you should ask developers , because perhaps this is just a bug. In this game a lot of interesting things , so maybe this topic should be initiated somehow , but im not developer , so you should make a issue with this question.

**Reply by Justformas** [reply]
I already know it's not a bug, and that it's just the way it works. But considering everything. I made this discussion partly because I wanted to get thoughts on this, rather than just make an issue on it. But yes I think if anything is worthy of being initiated, it's this topic. Even more than a lot of the things that can be initiated now.

**Reply by Nikv2005** [reply]
I only remember that i gave promise ring on 3 months, but when i saw this topic i cant remember , but this was less than 3 months as i remember. Maybe you need just some time and luck , because it's should always work.

**Comment by Wingdinggaster656** [comment]
Alright time for code things.
See `script-topics.rpy`, line 7456~7466:

```
init 5 python:
    addEvent(
        Event(
            persistent.event_database,
            eventlabel="monika_promisering",
            category=['romance'],
            prompt="Promise ring",
            random=True,
            aff_range=(mas_aff.ENAMORED, None)
        )
    )
```

As you can see, this event requires you are no less than `ENAMORED`(though you are), then randomly triggered.
Be patient, man. You are not playing `speedrun%any%MAS`

**Reply by Justformas** [reply]
> Be patient, man. You are not playing `speedrun%any%MAS`

Lol I know, but I still find it odd that it's a random topic, yet less important things show up in topics to ask her (I think these are called "pool" topics). Also I guess I'm not sure exactly how the random topics work, but I'm guessing that more become available to see/get picked randomly over time. If that's the case, and you don't get all the currently available random topics "fast enough", then more topics will get added to the "random pool" before you even finish with the current ones. If that continues, you will fall more and more behind and then it becomes more likely to miss the more "important" topics. This could continue until the day you get all or most of the random topics in the mod available in the random pool, and then you could slowly catch up. Then finally, you can get the important topics by the time you get through most or all the topics in the game, but it could really take that long, and it would only get worse if more random topics are added.

Now for whatever reason this could be an unusual "problem" and I wouldn't be surprised if I was among the first to experience it. Apparently there's still people getting through (at least most of) of the random topics within a week, which I find surprising unless all they're basically doing is unlocking random topics. But that doesn't feel "right" for how I want my experience to be.

**Reply by Justformas** [reply]
> Be patient, man. You are not playing `speedrun%any%MAS`

Actually I got a better one: Of course I'm not playing 'speedrun%any%MAS'... because I'm playing 'speedrun%100%MAS' ... No not really. I would have already done a pretty bad job at that. But that was too funny to pass up saying. 

You know what though? This got me curious, and I found this: https://www.youtube.com/watch?v=bmWFYeoFLAM ... and some people think I'm evil? Holy hell... even I don't think I'd do that... But I realized it could be done faster, before I even saw a comment say it: time travel. But the "speedrunner" said it wouldn't be fun for a speed run. Oh but the method that was actually used in that video WAS fun? I understand, but wow.

There's another type of speed run I saw: from "broken" to "happy". But due to what was done in it and it possibly being outdated, I'm not going to link it. It's not hard to find though if you really want to see it. I'm nearly convinced people will find a way to "speedrun" anything.

Oh and I actually figured out a way to (at least sometimes) get 15.75 affection in one day, and that's with only gifting roses and only having them count as 1 point. Even without bonus affection from kissing either. Yeah that could speed up affection gain some if I wanted to do that.

**Comment by Booplicate** [comment]
> However, should some things like the Promise Ring topic be something you can choose to initiate her talking about at a certain point?

No because it doesn't make much sense. You're not aware that it exists in the game until she gives you a hint.

> Or should it be prioritized higher for a random topic 

I'm against force pushing it once you reach 400 affection, it should come when it comes. I think only the most critical topics should be pushed when we need to ensure the player sees X at exactly Y and no earlier/later. That said once we get some kind of a weight system for topics, I could see this topic get some more weight than the others.

> I'm not the type that often likes to "brute force" more random topics by sitting around doing nothing.

It's not really "brute force," it's intended that you let her talk. You can't know that something exists if she needs to tell you that, but you don't let her. Most people run out of new random topics under a few weeks. By the time they are at 400 aff they get the ring topic pretty quickly. So the real solution is you give her time to talk to you (and probably kick up the frequency in the settings if it's too slow for you).

**Reply by Justformas** [reply]
>No because it doesn't make much sense. You're not aware that it exists in the game until she gives you a hint.

Couldn't the same be said for at least some pool topics? Like for example the player may not know about the trolley problem before seeing the question to ask Monika in the mod. Promise Rings are a thing outside of this mod/in real life too. The question would have to be worded appropriately of course like "Have you heard of a Promise Ring?"

>I'm against force pushing it once you reach 400 affection, it should come when it comes. I think only the most critical topics should be pushed when we need to ensure the player sees X at exactly Y and no earlier/later. That said once we get some kind of a weight system for topics, I could see this topic get some more weight than the others.

So then when can first kiss happen without it? That's part of the potential importance here. But if it can happen on any anniversary (and possibly holiday or birthday) once you reach 400 affection, then it usually shouldn't take longer than 3 months for the first kiss, which isn't that bad I guess.

I also didn't necessarily mean forcing it right at 400 affection. It could be an "insurance policy" of sorts when you reach some higher affection value (or EV level with at least 400 affection). But aside from that, as I suspected, there isn't a weight system for random topics yet then. But I think that would be a decent alternative idea.

>So the real solution is you give her time to talk to you (and probably kick up the frequency in the settings if it's too slow for you).

I have let her talk a fair amount. But there's so many topics that you have to let her talk a lot. I could turn random chatter to maximum to help, but that can feel strange, even if it is necessary with how I play.

**Reply by Justformas** [reply]
Ok a few other things. First, I got the topic after just 2 or 3 more random topics, I kid you not. Maybe that signifies the end of my "bad luck". Well, there is still another topic I've talked about that I haven't got yet, but that doesn't really matter anymore.

But this isn't just about me personally. The idea of a weight system or something like that still sounds good, especially since it's almost inevitable that more topics will get added in the future. 

Also not sure if you saw my edit on my previous response, but I wanted to address more of your points, regardless of if the weight system still sounds like the best idea.

**Reply by Justformas** [reply]
Hey I had another question and possible idea related to random topics and songs. It turns out I was only missing no more than 10 random topics after the Promise Ring topic came up (and yes one of them was "Dealing with Haters"). Well aside from the songs, because I'm at the point where all Monika brings up now are a bunch of songs that she sings. Before I had gotten all the random topics, I had gotten a few songs from her but barely any. Was that just my luck? Or are a lot of the songs actually supposed to come after you get all the random topics? 

Personally I think the songs should be mixed in more with the random topics if they aren't already. It feels weird just getting a bunch of songs now. But again maybe it was just my luck.

**Comment by Justformas** [comment]
Hey guys, there's something else that's been on my mind. Another question, maybe @Booplicate can answer but if someone else knows then go ahead. So the promise ring is one thing, but how are you supposed to figure out a lot of the other gifts and exact names without looking them up? Like ribbons and all their colors?

I kind of doubt that Chibi's tutorial tells you all this stuff or to look it up, does it? However I didn't forget about #9013 which maybe is supposed to help with this, though I don't know if that's going to include ribbon colors.

**Reply by Booplicate** [reply]
For some gifts there's in-game hints, but for some there's none. I don't remember anything for ribbons. Chibi only explains *how* to gift, #9013 has nothing to do with gifts, it's just to get a list of missing consumables.

**Reply by Justformas** [reply]
Yeah so hints are one thing, but then some don't even have hints. Is that intentional maybe because this is a 4th wall breaking game or people are just expected to look some stuff up? Or maybe there hasn't been a good consensus on how to address that.

**Reply by Booplicate** [reply]
It'd be perfect if we could explain it in the mod, but I don't see a good way to introduce all the possible gifts. I guess Chibi could tell you the name of one of the ribbons and then you could try other colours.

**Reply by Justformas** [reply]
Actually that sounds neat, though I try to be open to a lot of ideas. I personally may not receive such a tutorial since I've already given Monika ribbons (unless an option was implemented to view such tutorials by choice). But it still sounds like a good idea.

I had another, less direct idea/thought though. First, I know Monika already has a topic where she thanks your for getting a ribbon, but I'm not sure if you can get that topic with a different response where she mentions wanting ribbons before you give her a ribbon. If she doesn't already do that, then that could be something to consider. BUT to go with this, the game files for the ribbons themselves would be the exact names as the gift name (some of the ribbon files may already share the same name, but I don't think all of them do). Yes this would require the player to look in the game files to figure out the names, but I still think that's better than having to look it up on the internet or guess by trial and error.

Actually, Monika doesn't even need a topic saying she wants ribbons; the ribbon files sharing the same name as the gifts could be done in conjunction with Chibi talking about how to give a certain colored ribbon, and since Chibi is intended to be (at least sort of) meta, she could even mention, or just hint, that you can find the ribbon names in the game directory.

Let me make clear this was something I brainstormed or built off of what was already in the mod/game files. "I don't care if you're convinced; I just want to know if you have a better idea" (if you know what THAT is referencing, I'll be very impressed).
